etl/steps/data/garden/state_capacity/2023-11-10/information_capacity_dataset.py

```diff
@@ -0,0 +1,126 @@
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = paths.load_dataset("information_capacity_dataset")
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["information_capacity_dataset"].reset_index()
+
+    #
+    # Process data.
+    #
+
+    # Drop country id columns
+    tb = tb.drop(columns=["ccodecow", "vdemcode"])
+
+    # Harmonize country names.
+    tb = geo.harmonize_countries(
+        df=tb, countries_file=paths.country_mapping_path, excluded_countries_file=paths.excluded_countries_path
+    )
+
+    # Reset index after dropping excluded countries
+    tb = tb.reset_index(drop=True)
+
+    # Add regional aggregations
+    tb = regional_aggregations(tb)
+
+    tb = tb.set_index(["country", "year"], verify_integrity=True)
+
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(
+        dest_dir, tables=[tb], check_variables_metadata=True, default_metadata=ds_meadow.metadata
+    )
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def regional_aggregations(tb: Table) -> Table:
+    """
+    Add regional aggregations for some of the indicators
+    """
+
+    tb_regions = tb.copy()
+
+    # Load population data.
+    tb_pop = paths.load_dataset("population")
+    tb_pop = tb_pop["population"].reset_index()
+
+    # Merge population data.
+    tb_regions = tb_regions.merge(tb_pop[["country", "year", "population"]], how="left", on=["country", "year"])
+
+    # List of index columns
+    index_cols = ["censusgraded_ability", "ybcov_ability", "infcap_irt", "infcap_pca"]
+
+    for col in index_cols:
+        # Create new columns with the product of the index and the population
+        tb_regions[col] = tb_regions[col].astype(float) * tb_regions["population"]
+
+    # Define regions to aggregate
+    regions = [
+        "Europe",
+        "Asia",
+        "North America",
+        "South America",
+        "Africa",
+        "Oceania",
+        "High-income countries",
+        "Low-income countries",
+        "Lower-middle-income countries",
+        "Upper-middle-income countries",
+        "European Union (27)",
+        "World",
+    ]
+
+    # Define the variables and aggregation method to be used in the following function loop
+    aggregations = dict.fromkeys(
+        index_cols + ["population"],
+        "sum",
+    )
+
+    # Add regional aggregates, by summing up the variables in `aggregations`
+    for region in regions:
+        tb_regions = geo.add_region_aggregates(
+            tb_regions,
+            region=region,
+            aggregations=aggregations,
+            countries_that_must_have_data=[],
+            frac_allowed_nans_per_year=0.1,
```

Comment:
@pabloarosado Do you know why the `frac_allowed_nans_per_year` is not working as expected? I would expect to see [here](http://staging-site-parriagadap/admin/variables/816330) more missing aggregations for Africa (too few countries covered), but I see it's not the case:
<img width="631" alt="image" src="https://github.com/owid/etl/assets/63430031/8c40b92a-038d-4d2e-b518-366a7561117d">

Reply:
Hey @paarriagadap can you give me a very specific example of what you would expect to have and what you are getting instead?

Reply:
I would expect to not see aggregations for Africa, because too many countries from the continent are covered:
<img width="627" alt="image" src="https://github.com/owid/etl/assets/63430031/960c6afa-f89c-4ea7-b872-0672ef65bf5f">

Reply:
For column "infcap_irt" and e.g. year 2000, I see that there are 7 African countries informed, which is more than 10% of all African countries (59). So, to me, it looks consistent. But if you have a specific example that sounds wrong, please let me know.

Reply:
Ah, you are right, what I said before would not explain the problem, sorry for the confusion.
I think here the problem is different: The function doesn't actually care about how many countries are in Africa. The function simply checks how many African countries appear in a given year. And, if 10% or more are nan, then it removes the aggregate. However, the data actually contains only 7 African countries, and all of them informed (at least for year 2000 column "infcap_irt", which is what I took as a reference). So actually 0% of the rows are nan, and therefore the aggregate is valid. Does that make sense?

Reply:
Ahh, that's it, thanks. I've always thought the parameter worked with all the countries, regardless if they are in the dataset (by looking at the regions table). I will try to find an alternative method then.

Reply:
I'm happy it's clear now. You could impose a list of countries that need to be informed, by passing `countries_that_must_have_data`. But maybe you can figure out a better solution. As I always mention, we should rethink our function to create aggregates at some point.